src/fetch.ts

```diff
@@ -79,15 +79,17 @@ export function fetchEventSource(input: RequestInfo, {
             }
         }
 
-        if (!openWhenHidden) {
+        if (typeof window !== 'undefined' && !openWhenHidden) {
             document.addEventListener('visibilitychange', onVisibilityChange);
         }
 
         let retryInterval = DefaultRetryInterval;
         let retryTimer = 0;
         function dispose() {
-            document.removeEventListener('visibilitychange', onVisibilityChange);
-            window.clearTimeout(retryTimer);
+            if (typeof window !== 'undefined' && !openWhenHidden) {

```

Comment:
same as [above](https://github.com/Azure/fetch-event-source/pull/28/files#r1081208467):

```suggestion
            if (typeof window !== 'undefined' && typeof document !== "undefined" && !openWhenHidden) {
```